src/dataprotection/azext_dataprotection/tests/latest/test_dataprotection_backup_vault.py

```diff
@@ -44,6 +44,34 @@ def test_dataprotection_backup_vault_create_and_delete(test):
             test.check('name', "{vaultName}")
         ])
         test.cmd('az dataprotection backup-vault delete -g "{rg}" --vault-name "{vaultName}" -y')
+        
+    @AllowLargeResponse()
+    @ResourceGroupPreparer(name_prefix='clitest-dpp-backupvault-', location='centraluseuap')
+    def test_dataprotection_backup_vault_create_with_cmk_update_and_delete(test):
+        test.kwargs.update({
+            'cmkKeyUri': "https://cmk-cli-test-keyvault.vault.azure.net/keys/cmk-cli-key1/24efffaddbe84838a1c39b6135edbdf5",
+            'cmkKeyUriUpdate': "https://cmk-cli-test-keyvault.vault.azure.net/keys/cmk-cli-key2/864fe3c0fcf14d75bd7d576a148ba51c",
+            'cmkUami': "/subscriptions/38304e13-357e-405e-9e9a-220351dcce8c/resourcegroups/clitest-dpp-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/cmk-cli-test-uami",
+        })
+        test.cmd('az dataprotection backup-vault create '
+                 '-g "{rg}" --vault-name "{vaultName}" -l "{location}" '
+                 '--storage-settings datastore-type="VaultStore" type="GeoRedundant" --type "UserAssigned" '
+                 '--user-assigned-identities {{"{cmkUami}":{{}}}} --cmk-encryption-key-uri "{cmkKeyUri}" '
+                 '--cmk-encryption-state Enabled --cmk-identity-type UserAssigned '
+                 '--cmk-infrastructure-encryption Enabled --cmk-user-assigned-identity-id  "{cmkUami}" ',
+                 checks=[
+                     test.check('name', "{vaultName}"),
+                     test.check('identity.type', "UserAssigned"),
+                     test.check('properties.securitySettings.encryptionSettings.state', "Enabled"),
+                     test.check('properties.securitySettings.encryptionSettings.infrastructureEncryption', "Enabled"),
+                     test.check('properties.securitySettings.encryptionSettings.keyVaultProperties.keyUri', "{cmkKeyUri}"),
+                     test.check('properties.securitySettings.encryptionSettings.kekIdentity.identityId', "{cmkUami}"),
+                     test.check('properties.securitySettings.encryptionSettings.kekIdentity.identityType', "UserAssigned")
+                 ])
+        test.cmd('az dataprotection backup-vault update -g "{rg}" --vault-name "{vaultName}" --cmk-encryption-key-uri "{cmkKeyUriUpdate}"', checks=[
```

Comment:
Add a check for switching identity to SystemAssigned and SystemAssigned,UserAssigned so that we can test all the flows sufficiently

Reply:
resolved

Reply:
Added a new test